src/test/java/roomescape/acceptance/AcceptanceTest.java

```diff
@@ -0,0 +1,9 @@
+package roomescape.acceptance;
+
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.annotation.DirtiesContext;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
```

Comment:
@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
Comment 는 어떤 원리로` SpringTestContainer 를 초기화해줄까요?`

Reply:
각 테스트마다 Context cache에서 ApplicationContext를 삭제하고 새로운 context에서 테스트가 실행되도록 합니다. 
>  for example, by modifying the state of a singleton bean, modifying the state of an embedded database, etc.
현재 Repository와 Acceptance 테스트에서 해당 어노테션을 사용하고 있는데, 싱글톤 빈들은 상태를 갖고 있지 않고 embedded db인 h2만 초기화하면 되는 상황입니다.

Context cache를 찾아보니 아래와 같이 테스트에서 context를 캐싱하는 것은 성능상 중요한 포인트였네요..!
> Support for the caching of loaded contexts is important, because startup time can become an issue — not because of the overhead of Spring itself, but because the objects instantiated by the Spring container take time to instantiate.

따라서 @DirtiesContext 대신 DB를 초기화해주는 DDL을 각 테스트 전에 실행하도록 변경하였습니다 :)